File: app/src/main/java/moe/haruue/wadb/component/home/HomeActivity.java
package moe.haruue.wadb.component.home;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.Lifecycle;

import moe.haruue.wadb.app.BaseActivity;

public class HomeActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (savedInstanceState == null) {
            Fragment fragment = new HomeFragment();
            getSupportFragmentManager().beginTransaction()
                    .replace(android.R.id.content, fragment)
                    .setMaxLifecycle(fragment, Lifecycle.State.RESUMED)
                    .commit();
        }
    }
}

File: app/src/main/java/moe/haruue/wadb/component/home/HomeActivity.kt
package moe.haruue.wadb.component.home

import android.os.Bundle
import androidx.lifecycle.Lifecycle
import moe.haruue.wadb.R
import moe.haruue.wadb.app.AppBarFragmentActivity

class HomeActivity : AppBarFragmentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        if (savedInstanceState == null) {
            val fragment = HomeFragment()
            supportFragmentManager.beginTransaction()
                    .replace(R.id.fragment_container, fragment)
                    .setMaxLifecycle(fragment, Lifecycle.State.RESUMED)
                    .commit()
        }
    }
}
